docs/proposed/psa-driver-interface.md

```diff
@@ -646,6 +646,36 @@ The core will not update the persistent state in storage while an entry point is
 
 In a multithreaded environment, the driver may only call these two functions from the thread that is executing the entry point.
 
+#### Built-in keys
+
+Opaque drivers may declare built-in keys. Built-in keys can be accessed, but not created, through the PSA Cryptography API.
+
+A built-in key is identified by its location and its **slot number**. Drivers that support built-in keys must provide a `"get_builtin_key"` entry point to retrieve the key data and metadata. The core calls this entry point when it needs to access the key, typically because the application requested an operation on the key. The core may keep information about the key in cache, and successive calls to access the same slot number should return the same data. This entry point has the following prototype:
+
+```
+psa_status_t acme_get_builtin_key(psa_drv_slot_number_t slot_number,
+                                  psa_key_attributes_t *attributes,
+                                  uint8_t *key_buffer,
+                                  size_t key_buffer_size);
```

Comment:
I am not sure but what about having a `key_buffer_length` parameter here? Probably better for key buffer handling consistency and if the buffer is too small the driver can just return the minimum required size and the core doesn't have to do any calculation.

Reply:
Agreed, that does make it more consistent with the convention described in psa-driver-interface.md.